src/init/starship.ps1

```diff
@@ -1,38 +1,79 @@
 #!/usr/bin/env pwsh
 
+function Get-Cwd {
+    $cwd = Get-Location
+    $provider_prefix = "$($cwd.Provider.ModuleName)\$($cwd.Provider.Name)::"
+    return @{
+        # Resolve the actual/physical path
+        # NOTE: ProviderPath is only a physical filesystem path for the "FileSystem" provider
+        # E.g. `Dev:\` -> `C:\Users\Joe Bloggs\Dev\`
+        Path = $cwd.ProviderPath;
+        # Resolve the provider-logical path 
+        # NOTE: Attempt to trim any "provider prefix" from the path string.
+        # E.g. `Microsoft.PowerShell.Core\FileSystem::Dev:\` -> `Dev:\`
+        LogicalPath =
+            if ($cwd.Path.StartsWith($provider_prefix)) {
+                $cwd.Path.Substring($provider_prefix.Length)
+            } else {
+                $cwd.Path
+            };
+    }
+}
+
+function Invoke-Native {
+    param($Executable, $Arguments)
+    # Build an arguments string which follows the C++ command-line argument quoting rules
+    # See: https://docs.microsoft.com/en-us/previous-versions//17w5ykft(v=vs.85)?redirectedfrom=MSDN
+    $Arguments = $Arguments | ForEach-Object {
+        $_ = $_ -Replace '(\\+)"','$1$1"' # Escape backslash chains immediately preceeding quote marks.
+        $_ = $_ -Replace '(\\+)$','$1$1'  # Escape backslash chains immediately preceeding the end of the string.
+        $_ = $_ -Replace '"','\"'         # Escape internal quote marks.
+        "`"$_`""                          # Quote the argument.
+    }
+    # Use the stop-parsing symbol (--%) to ensure the argument string is passed along unchanged 
+    # See: https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_parsing?view=powershell-5.1
+    Invoke-Expression "& `"$Executable`" --% $($Arguments -Join ' ')"
+}
```

Comment:
Some of the stuff that broke long paths appears to be happening in the powershell part. Here's a fix:
```suggestion
function Invoke-Native {
    param($Executable, $Arguments)

    # Build an arguments string which follows the C++ command-line argument quoting rules
    # See: https://docs.microsoft.com/en-us/previous-versions//17w5ykft(v=vs.85)?redirectedfrom=MSDN
    $Arguments = $Arguments | ForEach-Object {
        $t = $_ -Replace '(\\+)"','$1$1"' # Escape backslash chains immediately preceeding quote marks.
        $t = $t -Replace '(\\+)$','$1$1'  # Escape backslash chains immediately preceeding the end of the string.
        $t = $t -Replace '"','\"'         # Escape internal quote marks.
        "`"$t`""                          # Quote the argument.
    }

    $process = New-Object System.Diagnostics.Process
    $process.startInfo.FileName = $Executable
    $process.startInfo.Arguments = $Arguments
    
    # Use UTF-8
    $process.startInfo.StandardOutputEncoding = [System.Text.Encoding]::UTF8
    # Required for UTF-8
    $process.startInfo.RedirectStandardOutput = $true

    $process.startInfo.CreateNoWindow = $true
    $process.startInfo.UseShellExecute = $false

    $process.Start() > $null

    $process.StandardOutput.ReadToEnd()
}
```

Reply:
Edit: with this approach to command execution, instead of your escape code, this worked:
```ps1
    foreach ($arg in $Arguments) {
        $process.startInfo.ArgumentList.Add($arg)
    }
```

Reply:
I tested your version of argument escaping, because it looks like `ArgumentList` doesn't work on PS5, but it looks like it doesn't work. Maybe try  implementing a solution based on something like https://github.com/PowerShell/PowerShell/pull/13482#issuecomment-678522514? (I'd still prefer using `ArgumentList` for PS 6.1 and up)

Because this all looks very broken without `ArgumentList` I think it might be best to just drop support for PS < 6.1 if you don't feel like doing that.

Reply:
`ProcessStartInfo.ArgumentList` was only added in .NET 5.0, so this code will only work in PowerShell 6+

https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.processstartinfo.argumentlist?view=net-5.0&viewFallbackFrom=netframework-4.8#System_Diagnostics_ProcessStartInfo_ArgumentList

(which sucks because I'd much rather not be doing this manual escaping)

Reply:
I've adjusted the `Invoke-Native` function to use `System.Diagnostics.ProcessStartInfo` as in your suggested change above. If `ArgumentList` is available we use it, otherwise we fall back to building an escaped argument string.

Do you have any examples of argument strings which break the manual escaping process? All my test sample strings work correctly:

```
$a = 'hello "world" \" \\" goodbye "moon"' 
$b = 'path\without\whitespace\'
$c = 'path\with whitespace\'
$d = '--flag=argument passed to flag'
$e = 'five trailing backslashes \\\\\'
$f = '\\\\\five leading backslashes'
$g = '\\?\C:\Windows\Extended Path Syntax\'
```

Reply:
It looks like your solution is sometimes appending `;` to the argument.
When I got rid of all the `;` in `ForEach-Object` it appeared to work fine. You also might want to avoid assigning to `$_`. The VSCode PowerShell extension complains about that.
Beyond that: `ArgumentList` was added in PowerShell 6.1 with .Net Core 2.1. I think only PowerShell 7.1 has .Net 5.

Reply:
Ah, there was a mistyped `;` I thought I'd already fixed, but I'd only fixed it in my testing script 🤦‍♂️.

Re: `ArgumentList` that makes sense. I was following the docs but I think I was in the "framework" docs not the "core" docs. I'm testing for the property, not the PowerShell version so it will work on PS 6 & 7.